lib/scope.js

```diff
@@ -2170,10 +2273,244 @@ module.exports = {
+      let fields = await scope.getAllFields( scope, { html: html });
+      let buttons = [];
+
+      // For every field on the page
+      for ( let field of fields ) {
+
+        // * @param fields - data representing one DOM field
+        // * @param fields[n].selector {str} - used to find the DOM node
+        // * @param fields[n].tag {str} - HTML tag of node
+        // * @param fields[n].type {str} - HTML type of node if it has one
+        // * @param fields[n].guesses {arr} - arr of objects that could
+        // *    possibly match a table row's `var` column
+        // * @param fields[n].guesses[n].var {str} - name of variable
+        // * @param fields[n].guesses[n].value {str} - value of a choice
+        // * @param <fields[n].guesses[n].trigger> {str} - var name that triggered this page
+
+        // If the field is a button, save it in a list of buttons that we'll deal with later
+        if ( field.tag === `button` ) {
+          buttons.push( field );
+        } else {
+          // Try to give a random answer for that element tag
+          await scope.set_random_input_for[ field.tag ]( scope, { field: field });
+          process.stdout.write(`\x1b[36m${ '*' }\x1b[0m`);  // assumes var was set if no error occurred
+        }
+
+      }  // End for all page fields
+
+      return buttons;
+
+    },  // Ends scope.steps.set_random_page_vars()
 
   },  // ends scope.steps
 
+  set_random_input_for: {
+    a: async function (scope, { fields }) {
+      // We don't do anything with links for random input tests
+    }, 
+    buttons: async function (scope, { fields }) {
+      // Get a random button
+      let field = faker.helpers.arrayElement( fields );
+      let handle = await scope.get_handle_from_selector( scope, { field: field });
+      // Send it to be tapped
+      await scope.funnel_the_answer( scope, { handle, field, set_to: `doesn't matter` });
+    },
+    textarea: async function (scope, { field }) {
+      // Get random paragraph of text
+      let answer = faker.lorem.paragraph();
+      // Get the element's puppeteer handler
+      let handle = await scope.get_handle_from_selector( scope, { field: field });
+      // Type the random text into the textarea
+      await scope.funnel_the_answer( scope, { handle, field, set_to: answer });
+      // await scope.enter_answer[ field.type ]( scope, { handle, set_to: answer } );
+    },
+    select: async function (scope, { field }) {
+      // Get the values of all the options of the field
+      let handle = await scope.get_handle_from_selector( scope, { field: field });
+      // Get the values of all the choices
+      let answer_choices = await handle.evaluate(function ( elem ) {
+        let values = Array.from(elem.querySelectorAll(`option`)).map(element=>element.value);
+        return values;
+      });
+
+      // Remove the "no selection" option from the list ('')
+      // Should we allow '' as a valid choice - no selection?
```

Comment:
IMO we shouldn't; only if it's an optional field, and that's def out of scope for MVP

Reply:
```suggestion
TODO: blue sky - for optional fields allow no selection
```